libs/sdk-common/src/invoice.rs

```diff
@@ -75,6 +82,25 @@ impl From<SystemTimeError> for InvoiceError {
     }
 }
 
+fn parse_short_channel_id(id_str: &str) -> InvoiceResult<u64> {
+    let parts: Vec<&str> = id_str.split('x').collect();
+    if parts.len() != 3 {
+        return Ok(0);
```

Comment:
I know this was already there, but shouldn't this return an error?

Reply:
Changed to throw an error, also added some tests